### PR TITLE
feat(docs): add datatable display and sort example

### PR DESCRIPTION
**Pull request by albertms10**
related to https://github.com/IBM/carbon-components-svelte/pull/356#issuecomment-715453911

---

- feat(docs): add datatable display and sort example
- fix(docs): string interpolation in display method body

### CODE REVIEW

File: docs/src/pages/components/DataTable.svx [hunk 1]
```diff
@@ -192,6 +192,73 @@ The slot name for the table header cells is `"cell-header"`.
...
 />
 
+### With custom display and sort methods
+
+<DataTable title="Load balancers" description="Your organization's active load balancers."
+  headers="{[
+    { key: "name", value: "Name" },
+    { key: "protocol", value: "Protocol" },
+    { key: "port", value: "Port" },
+    { key: "cost", value: "Cost", display: (cost) => `${cost} €` },
```
Comment (metonym) [inline docs/src/pages/components/DataTable.svx:202]:
```suggestion
    { key: "cost", value: "Cost", display: (cost) => cost + " €" },
```

File: docs/src/pages/components/DataTable.svx [hunk 2]
```diff
@@ -192,6 +192,73 @@ The slot name for the table header cells is `"cell-header"`.
   ]}"
 />
 
+### With custom display and sort methods
+
+<DataTable title="Load balancers" description="Your organization's active load balancers."
```
Comment (metonym) [inline docs/src/pages/components/DataTable.svx:197]:
```suggestion
<DataTable sortable title="Load balancers" description="Your organization's active load balancers."
```

Add the `sortable` prop to demo the custom sort method

**Review by metonym — APPROVED** [review]
Excellent

File: docs/src/pages/components/DataTable.svx [hunk 3]
```diff
@@ -192,6 +192,73 @@ The slot name for the table header cells is `"cell-header"`.
...
+  headers="{[
+    { key: "name", value: "Name" },
+    { key: "protocol", value: "Protocol" },
+    { key: "port", value: "Port" },
+    { key: "cost", value: "Cost", display: (cost) => cost + " €" },
+    {
+      key: "expireDate",
+      value: "Expire date",
+      display: (date) => new Date(date).toLocaleString(),
+      sort: (a, b) => (new Date(a) <= new Date(b) ? -1 : 1),
```
Comment (albertms10) [inline docs/src/pages/components/DataTable.svx:207]: The date comparison operation may be simplified with `new Date(a) - new Date(b)` for better results.

fixed in #382

**Pull request merged**